src/layers/ethernet/mod.rs

```diff
@@ -84,6 +85,33 @@ impl Layer for Ethernet {
         }
     }
 
+    fn encode_bytes(
+        &self,
+        next_layer: Option<&Vec<u8>>,
+        info: Option<&str>,
+    ) -> Result<Vec<u8>, Error> {
+        let mut result = vec![];
+
+        result.extend(self.dst_mac.as_slice());
+        result.extend(self.src_mac.as_slice());
+
+        let ethertype: u16 = match info {
```

Comment:
This could be much better, create inverses of the static maps maybe?

Reply:
Yes, every `layer` should similarly have an _inverse_ map for sculpting (which is populated and updated using `register_defaults`).

Reply:
I think this can be implemented in the next version of this commit. As you said above, in the interest of making this reviewable, I want to push these changes once the current once have been approved and merged.